esmf_regrid/schemes.py

```diff
@@ -575,29 +1051,186 @@ def __call__(self, cube):
             area-weighted regridding via :mod:`esmpy` generated weights.
 
         """
-        src_x, src_y = (_get_coord(cube, "x"), _get_coord(cube, "y"))
-
-        # Check the source grid matches that used in initialisation
-        if self.src_grid != (src_x, src_y):
-            raise ValueError(
-                "The given cube is not defined on the same "
-                "source grid as this regridder."
-            )
+        if cube.mesh is not None:
+            # src_mesh = cube.mesh
+            # location = cube.location
+            # if self.src != (src_mesh, location):
+            #     raise ValueError(
+            #         "The given cube is not defined on the same "
+            #         "source mesh as this regridder."
+            #     )
+            dims = [cube.mesh_dim()]
 
-        if len(src_x.shape) == 1:
-            grid_x_dim = cube.coord_dims(src_x)[0]
-            grid_y_dim = cube.coord_dims(src_y)[0]
         else:
-            grid_y_dim, grid_x_dim = cube.coord_dims(src_x)
+            src_x, src_y = (_get_coord(cube, "x"), _get_coord(cube, "y"))
```

Comment:
since the next line checks that these are equal to the coords in self.src, I feel feel like using the same name is appropriate. Either way, the fact that these represent the x and y coordinates on the source (as opposed to the target) seems more important to get across than the fact they derive from the cube given as an argument in `__call__`

Reply:
Yes I see your point that it's important to retain the information that is currently included. But I still feel that it's easier to understand what's going on if `src_x` and `src_y` are also shown to be different from `self.src`.

`input_src_x`? `new_src_x`? `cube_src_x`?